src/display_gadfly.jl

```diff
@@ -33,20 +33,30 @@ initialized(pzc::PanZoomCallbacks) = pzc.idpzk != 0
 
 # While figures are by default associated with Windows (each with a
 # single Canvas), you can have multiple figures per window.
-type Figure
+type Figure <: Gtk.GtkBox
+    handle::Ptr{Gtk.GObject}
     canvas::GtkCanvas
+    tb::Gtk.GtkToolbar
     prepped         # tuple, a pre-processed Plot to speed rendering
     cc::Compose.Context   # fully-rendered Plot (useful for hit-testing)
     tweaks::Dict{Tuple{Symbol,Symbol},Any}  # post-rendering property changes
     panzoom_cb::PanZoomCallbacks
     figno::Int
 
-    function Figure(c::GtkCanvas, p::Plot)
+    function Figure(p::Plot)
+    
```

Comment:
Dumb point, but I'd prefer not to have a blank line at the beginning of functions